lib/seccomp/CSystemCallFilter_MacOSX.cc

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+#include "seccomp/CSystemCallFilter.h"
+
+#include <core/CLogger.h>
+
+#include <paths.h>
+#include <cstdio>
+#include <fstream>
+#include <sandbox.h>
+
+namespace ml {
+namespace seccomp {
+
+// The Sandbox rules deny all actions apart from creating fifos,
+// opening files and reading and writing.
+// (allow file-write*) is required for mkfifo and that permission
+// can not be set using the more granular controls.
+static const std::string SANDBOX_RULES("\
+    (version 1) \
+    (deny default) \
+    (debug all) \
+    (allow file-read*) \
+    (allow file-read-data) \
+    (allow file-write*) \
+    (allow file-write-data)");
+
+std::string getTempDir() {
+    // In production this needs to match the setting of java.io.tmpdir.  We rely
+    // on the JVM that spawns our controller daemon setting TMPDIR in the
+    // environment of the spawned process.
+    const char* tmpDir(::getenv("TMPDIR"));
```

Comment:
consider using boost for that, it handles some more cases afaik and can be easily replaced with std once we switch to C++17

Reply:
I'm not sure what you mean is there a boost version of getenv? I'm using the temp dir set by elasticsearch

Reply:
I mean ` boost::filesystem::temp_directory_path()`